tests/ActivityTest.php

```diff
@@ -43,26 +43,17 @@ public function testInstantiation() {
     }
 
     public function testFromJSONInvalidNull() {
-        $this->setExpectedException(
-            'InvalidArgumentException',
-            'Invalid JSON: ' . JSON_ERROR_NONE
-        );
+        $this->setExpectedException('TinCan\JSONParseErrorException');
```

Comment:
Yeah, I don't mind it, though in the newer parlance there is `expectExceptionCode` and `expectExceptionMessage` which would cover the same assertions. In general this seems a very divisive issue, whether to test the underlying parts of the Exception, or just for that specific Exception, and my personal feeling goes back and forth on the matter. Given that PHP seems to commonly change its own messages I think in this platform we are just as well off not checking the specific properties, as sad as that is from a public interface.

FWIW I drew some inspiration from https://thephp.cc/news/2016/02/questioning-phpunit-best-practices which is authored by the primary author/maintainer of PHPUnit. In particular his statement around multiple exceptions in a single test case.